app/lib/task/scheduler.dart

```diff
@@ -206,8 +207,40 @@ Future<void> schedule(
             arguments: [payload!],
             description: description!,
           );
+        } on ZoneExhaustedException catch (e, st) {
+          // A zone being exhausted is normal operations, we just use another
+          // zone for 15 minutes.
+          _log.info(
+            'zone resources exhausted, banning ${e.zone} for 30 minutes',
+            e,
+            st,
+          );
+          // Ban usage of zone for 30 minutes
+          zoneBannedUntil[e.zone] = clock.now().add(Duration(minutes: 30));
+
+          rollbackPackageState = true;
```

Comment:
I wish we had a better way to do this sort of thing...

It's almost tempting to just have `} on Exception catch (e, st) {`
and then use `if (e is ZoneExhaustedException)` instead of  `} on ZoneExhaustedException catch (e, st) {`.

That would provide a more reasonable control flow.